b2share/modules/remotes/views.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+# Copyright (C) 2016 University of Tuebingen, CERN.
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""B2Share remotes (b2drop) integration REST API."""
+
+from __future__ import absolute_import
+
+from urllib.parse import urlparse
+
+from flask import Blueprint, jsonify, abort, session, request, url_for, current_app
+from invenio_rest import ContentNegotiatedMethodView
+
+from invenio_files_rest.serializer import json_serializer
+
+from .errors import UserError
+from .b2drop import B2DropClient
+
+
+blueprint = Blueprint(
+    'remotes',
+    __name__,
+)
+
+
+def dict_to_json_serializer(dictionary, code=200, headers=None):
+    response = jsonify(dictionary)
+    response.status_code = code
+    if headers is not None:
+        response.headers.extend(headers)
+    return response
+
+
+class RemoteList(ContentNegotiatedMethodView):
+    view_name = 'remotes'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(RemoteList, self).__init__(
+            serializers={
+                'application/json': dict_to_json_serializer,
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def put(self, service):
+        if service == 'b2drop':
+            auth = request.get_json()
+            config = current_app.config.get('B2DROP_SERVER')
+            b2drop_client = B2DropClient(username=auth.get('username'),
```

Comment:
@thijscobben We would need an OAuth server on B2Drop to avoid this nasty hack.
Where can we create issues for B2Drop? I see many projects in the B2Drop github organization.